### PR TITLE
Harfbuzz update to 2.7.2, use meson

### PR DESCRIPTION
**Pull request by satmandu**
Part of https://github.com/skycocker/chromebrew/pull/4450 and upcoming adwaita_icon_theme PR

There MAY be circular dependencies here...   Do we need a harfbuzz_sub package?

Works properly:
- [x] x86_64

### CODE REVIEW

**Comment by satmandu** [comment]
HOLD this please. This may need newer graphite installed...

File: packages/harfbuzz.rb [hunk 1]
```diff
@@ -3,34 +3,37 @@
...
-  depends_on 'six' => :build
+  depends_on 'six' => ':build'
+  depends_on 'meson' => ':build'
+#  depends_on 'graphite' => ':build'
 
   def self.build
-    system "./configure --prefix=#{CREW_PREFIX} --libdir=#{CREW_LIB_PREFIX}"
-    system "make"
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
```
Comment (uberhacker) [inline packages/harfbuzz.rb:23]: Does this break arm builds?

File: packages/harfbuzz.rb [hunk 2]
```diff
@@ -3,34 +3,37 @@
...
-  depends_on 'glib'
+#  depends_on 'cairo' => ':build'
+  depends_on 'glib' => ':build'
+#  depends_on 'gobject_introspection' => ':build'
+  depends_on 'ragel' => ':build'
   depends_on 'freetype_sub'
-  depends_on 'six' => :build
+  depends_on 'six' => ':build'
+  depends_on 'meson' => ':build'
+#  depends_on 'graphite' => ':build'
```
Comment (uberhacker) [inline packages/harfbuzz.rb:19]: Please remove the single quotes around `:build`.

File: packages/harfbuzz.rb [hunk 3]
```diff
@@ -3,34 +3,37 @@
...
 
   def self.build
-    system "./configure --prefix=#{CREW_PREFIX} --libdir=#{CREW_LIB_PREFIX}"
-    system "make"
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
+    system "meson",
+      "-Dprefix=#{CREW_PREFIX}",
+      "-Dlibdir=#{CREW_LIB_PREFIX}",
```
Comment (uberhacker) [inline packages/harfbuzz.rb:26]: Use `#{CREW_MESON_OPTONS}`?

File: packages/harfbuzz.rb (hunk 2 above)
Comment (satmandu) [reply packages/harfbuzz.rb]: done

File: packages/harfbuzz.rb (hunk 3 above)
Comment (satmandu) [reply packages/harfbuzz.rb]: done

File: packages/harfbuzz.rb (hunk 1 above)
Comment (satmandu) [reply packages/harfbuzz.rb:23]: Doesn't seem to be a problem with gcc10.

File: packages/harfbuzz.rb [hunk 4]
```diff
@@ -3,34 +3,36 @@
...
-  })
 
-  depends_on 'glib'
+#  depends_on 'cairo' => ':build'
+  depends_on 'glib' => :build
+#  depends_on 'gobject_introspection' => ':build'
+  depends_on 'ragel' => :build
   depends_on 'freetype_sub'
   depends_on 'six' => :build
+  depends_on 'meson' => :build
```
Comment (uberhacker) [inline packages/harfbuzz.rb:18]: meson is already included in "core".

Comment (satmandu) [reply packages/harfbuzz.rb]: fixed

**Pull request merged**